cset-workflow/meta/rose-meta.conf

```diff
@@ -115,697 +232,1353 @@ sort-key=met1
+
+    strftime format strings are supported, and will be replaced with the
+    desired case study date or trial time. E.g: `/data/%Y%m%d/model1/*.nc`
 
-[template variables=SUBAREA_LON_BOUND_LEFT]
-ns=Diagnostics
-description=Left edge coordinate of the sub-area, real.
-help=Recommend looking at the input data to get these values. Uses the grid's native units.
-type=real
-sort-key=subsection2
+    For forecast lead time, %N can be used which is replaced by the 3 digit
+    lead time in hours. The regular placeholders are still usable, and refer
+    to the initiation time E.g: `/data/case_%Y%m%d/model1_%N.nc`
+
+    Commonly useful placeholders:
+    %Y: Year, e.g: 2024
+    %m: Month, e.g: 12
+    %d: Day, e.g: 31
+    %H: Hour, e.g: 23
+    %M: Minute, e.g: 59
+    %N: Forecast lead time in hours, e.g: 012
+
+    For more on strftime format strings, see: https://strftime.org
+type=quoted
 compulsory=true
+sort-key=a2
 
-[template variables=SUBAREA_LON_BOUND_RIGHT]
-ns=Diagnostics
-description=Right edge coordinate of the sub-area, real.
-help=Recommend looking at the input data to get these values. Uses the grid's native units.
-type=real
-sort-key=subsection2
+[template variables=m10_date_type]
+ns=Models and Cases/Model 10
+title=Date placeholder type
+description=Type of date templated into the data path.
+help=The type of date templated into the data path. Affects interpretation of
+    strftime placeholders in the data path.
+
+    Forecast initiation time, AKA data time, is the time of the validity time of
+    the first data point. For realtime forecasts this is approximately when the
+    forecast was started.
+
+    Validity time is when the data is predicting for.
+
+    Forecast lead time is how far from the forecast initiation time the data is.
+values="initiation", "validity", "lead"
+value-titles=Forecast Initiation Time, Validity Time, Forecast Lead Time
 compulsory=true
+trigger=template variables=m10_data_period: "validity", "lead";
+sort-key=b1
 
-[template variables=DETERMINISTIC_PLOT_INFLOW_PROPERTIES]
-ns=Diagnostics
-description=Extracts data required for, and calculates the inflow properties diagnostic, plotting on a map.
-            Required STASH m01s20i119, m01s00i025, m01s00i033.
-help=See includes/deterministic_plot_inflow_properties.cylc
-type=python_boolean
+[template variables=m10_data_period]
+ns=Models and Cases/Model 10
+title=Data period
+description=The period of the underlying data
+help=The period of the input data in each file as an ISO 8601 duration. This
+    is used to fetch files at regular validity or lead times.
+    E.g. for hourly data: PT1H
+type=quoted
 compulsory=true
+sort-key=b2
 
-[template variables=DOMAIN_HISTOGRAM_SERIES]
-ns=Diagnostics
-description=Create a series of histogram plots for selected vertical levels for each cycle time.
-help=See includes/deterministic_domain_histogram_series.cylc
-trigger=template variables=HISTOGRAM_TYPE: True;
+[template variables=m10_preprocessing]
```

Comment:
I suggest we open a new ticket for later making this more flexible rather than hardcoding 10 models. Not a high priority at this point though

Reply:
I've opened https://github.com/MetOffice/CSET/issues/817 for this.